mlrun/datastore/targets.py

```diff
@@ -1429,6 +1431,215 @@ def as_df(
+        """add storey writer state to graph"""
+        self.add_writer_step(graph, after, features, key_columns, timestamp_key)
+
+    def get_table_object(self):
+        from storey import MongoDBDriver, Table
+
+        # TODO use options/cred
+        (
+            endpoint,
+            db,
+            collection,
+        ) = self._parse_url()
+        return Table(
+            f"{db}/{collection}",
+            MongoDBDriver(webapi=endpoint),
+            flush_interval_secs=mlrun.mlconf.feature_store.flush_interval,
+            support_aggr=False,
+        )
+
+    def add_writer_step(
+        self,
+        graph,
+        after,
+        features,
+        key_columns=None,
+        timestamp_key=None,
+        featureset_status=None,
+    ):
+        key_columns = list(key_columns.keys())
+        column_list = self._get_column_list(
+            features=features, timestamp_key=timestamp_key, key_columns=key_columns
+        )
+        table = self._resource.uri
+        graph.add_step(
+            name=self.name or "MongoDBTarget",
+            after=after,
+            graph_shape="cylinder",
+            class_name="storey.NoSqlTarget",
+            columns=column_list,
+            header=True,
+            table=table,
+            index_cols=key_columns,
+            # storage_options=self._get_store().get_storage_options(),
+            **self.attributes,
+        )
+
+    def as_df(
+        self,
+        columns=None,
+        df_module=None,
+        entities=None,
+        start_time=None,
+        end_time=None,
+        time_column=None,
+        **kwargs,
+    ):
+        try:
+            query = kwargs["query"]
+        except KeyError:
+            query = {}
+        if time_column:
+            time_query = {time_column: {}}
+            if start_time:
+                time_query[time_column]["$gte"] = start_time
+            if end_time:
+                time_query[time_column]["$lt"] = end_time
+            if time_query[time_column] and query:
+                query.update(time_query)
+            elif time_query[time_column] and not query:
+                query = time_query
+
+        from pymongo import MongoClient
+
+        connection_string, db_name, collection_string = self._parse_url()
+        mongodb_client = MongoClient(connection_string)
+        db = mongodb_client[db_name]
+        collection = db[collection_string]
+        cursor = collection.find(query)
+
+        df = pd.DataFrame(list(cursor))
+        if "_id" in df.columns:
+            df["_id"] = str(df["_id"])
+        return df
+
+    def write_dataframe(
+        self, df, key_column=None, timestamp_key=None, chunk_id=0, **kwargs
+    ):
+        if hasattr(df, "rdd"):
+            raise ValueError("Spark is not supported")
+        else:
+            from pymongo import MongoClient
+
+            connection_string, db_name, collection_string = self._parse_url()
+            mongodb_client = MongoClient(connection_string)
```

Comment:
shouldn't you keep the client or you are writing just one time? Also, shouldn't you close the client (I'm just making sure you checked these stuff).